tests/unit/splinter_test.c

```diff
@@ -623,6 +625,58 @@ CTEST2(splinter, test_lookups)
    trunk_shadow_deinit(&shadow);
 }
 
+/*
+ * -----------------------------------------------------------------------------
+ * Simple test cases to exercise print / diagnostic functions provided
+ * by various sub-systems.  Test is now readily useful as an educational tool.
+ *
+ * NOTE: This test case is mentioned in external docs. Be careful what
+ *  changes you bring in here.
+ * -----------------------------------------------------------------------------
+ */
+CTEST2(splinter, test_splinter_print_diags)
+{
+   allocator *alp = (allocator *)&data->al;
+
+   trunk_handle *spl = trunk_create(data->splinter_cfg,
+                                    alp,
+                                    (cache *)data->clock_cache,
+                                    data->tasks,
+                                    test_generate_allocator_root_id(),
+                                    data->hid);
+   ASSERT_TRUE(spl != NULL);
+
+   uint64 num_inserts = splinter_do_inserts(data, spl, FALSE, NULL);
+   ASSERT_NOT_EQUAL(0,
+                    num_inserts,
+                    "Expected to have inserted non-zero rows, num_inserts=%lu",
+                    num_inserts);
+
+   platform_default_log("**** Splinter Diagnostics ****\n"
+                        "Generated by %s:%d:%s ****\n",
+                        __FILE__,
+                        __LINE__,
+                        __FUNCTION__);
+
+   trunk_print_super_block(Platform_default_log_handle, spl);
+
+   trunk_print_space_use(Platform_default_log_handle, spl);
+
+   platform_default_log("\n** trunk_sprint() **\n");
```

Comment:
Again here...

Reply:
fixed. [ really, sorry ! I did think I had grepped for all instances of `_sprint_` before publishing this reworked review. ]